manager/allocator/networkallocator/networkallocator.go

```diff
@@ -96,6 +80,15 @@ type NetworkAllocator interface {
 	// DeallocateTask releases all the endpoint resources for all the
 	// networks that a task is attached to.
 	DeallocateTask(t *api.Task) error
+
+	// AllocateLBAttachment Allocates a load balancer endpoint
+	AllocateLBAttachment(node *api.Node, nid string) error
```

Comment:
I think we have a problem if this interface keeps growing. Are we at the right level of abstraction?

Reply:
Technically, the interface is not growing.  I just renamed AllocateNode to AllocateLBAttachment.
Given your other comment about "attachment" in api/objects.proto, I'm going to leave the name as AllocateNode in the next PR.
// AllocateNode allocates the IP addresses for the networks to which		
// the node is attached.
AllocateNode(node *api.Node, nid string) error

Thanks again for the feedback.

Reply:
Were these previously doing attachment allocations?

Reply:
Just to be clear I'm going to remove AllocateLBAttachment and go back to AllocateNode.  So, the interface won't have any additional members.  Just the method signature will change.

Reply:
@pradipd I'm just wondering if we need a bit of refactoring here before making changes.

Reply:
Yes. These were previously doing attachment allocations.  Before it was only doing attachment for the ingress network.  Now it is doing attachment allocations for all overlay networks.

As for refactoring:  IMHO, the interface/abstraction seems correct to me.  The NetworkAllocator handles network resources across the cluster.  I.e. what network resource are needed for a task, service,  node, network.

On the other hand, network allocator is the only allocator in the package.   So, perhaps the abstraction looks off since there is only one resource being managed.  Perhaps, if there were other resources (maybe a CPUallocator or MemoryAllocator), the abstraction would make more sense.  E.g. I can easily see func (a *Allocator) allocateService (network.go:930) being updated in the future to also allocate other resource (CPU, Memory, etc.).

I don't really have a great answer on whether it's worth refactoring or not.  The current abstraction make immediate sense to me.